python/lsst/ts/m2gui/utility_monitor.py

```diff
@@ -0,0 +1,436 @@
+
+        self._report_temperatures()
+
+        self._report_displacements()
+
+    def _report_temperatures(self):
+        """Report the temperatures."""
+
+        for temperature_group in TemperatureGroup:
+            sensors = self.get_temperature_sensors(temperature_group)
+
+            temperatures = list()
+            for sensor in sensors:
+                temperatures.append(self.temperatures[sensor])
+
+            self.signal_utility.temperatures.emit((temperature_group, temperatures))
+
+    def get_temperature_sensors(self, temperature_group):
+        """Get the temperature sensors in a specific group.
+
+        Parameters
+        ----------
+        temperature_group : enum `TemperatureGroup`
+            Temperature group.
+
+        Returns
+        -------
+        `list [str]`
+            List of the sensors.
+        """
+
+        sensors = list()
+        for sensor in list(self.temperatures.keys()):
+            if sensor.startswith(temperature_group.name):
+                sensors.append(sensor)
+
+        return sensors
+
+    def _report_displacements(self):
+        """Report the displacements."""
+
+        for direction in DisplacementSensorDirection:
+            sensors = self.get_displacement_sensors(direction)
+            displacements = [self.displacements[sensor] for sensor in sensors]
+
+            self.signal_utility.displacements.emit((direction, displacements))
+
+    def get_displacement_sensors(self, direction):
+        """Get the displacement sensors in a specific direction.
+
+        Parameters
+        ----------
+        direction : enum `DisplacementSensorDirection`
+            Direction of the displacement sensors.
+
+        Returns
+        -------
+        `list [str]`
+            List of the sensors.
+        """
+
+        sensors = []
+        for sensor in self.displacements.keys():
+            result = re.match(rf"\AA\d-{direction.name}_z", sensor)
+            if result is not None:
+                sensors.append(result.group())
+
+        return sensors
+
+    def update_power(self, power_type, new_voltage, new_current):
+        """Update the power data.
+
+        Parameters
+        ----------
+        power_type : enum `PowerType`
+            Power type.
+        new_voltage : `float`
+            New voltage value in volt.
+        new_current : `float`
+            New current value in ampere.
+
+        Raises
+        ------
+        `ValueError`
+            Not supported power type.
+        """
+
+        if power_type == PowerType.Motor:
+            power = self.power_motor
+            signal_name = "power_motor"
+        elif power_type == PowerType.Communication:
+            power = self.power_communication
+            signal_name = "power_communication"
+        else:
+            raise ValueError(f"Not supported power type: {power_type!r}.")
+
+        tor = self._get_tor()
+        if (
+            abs(power["voltage"] - new_voltage) >= tor
```

Comment:
this can be vastly improved by creating e.g. a function taking *args as couples of new and old values:
```
def _has_changed(self, *pairs):
    tolerance = self._get_tor()
    for p in pairs:
        if abs(p[0] - p[1]) >= tor:
            return True
    return False

....

if self._has_changed((power["voltage"], new_voltage), (power["current"] - new_current)):
```

Reply:
Fixed in the commit: [2927fa9](https://github.com/lsst-ts/ts_m2gui/pull/7/commits/2927fa94e6b229312b4685c13217911ee3dca4f0). Thanks!